app/forms/provider_interface/provider_user_providers_form.rb

```diff
@@ -0,0 +1,52 @@
+module ProviderInterface
+  class ProviderUserProvidersForm
+    include ActiveModel::Model
+    attr_accessor :current_provider_user, :provider_user, :provider_ids
+
+    validate :at_least_one_provider_is_selected
+
+    def self.from_provider_user(attributes)
+      form = new(attributes)
+      form.provider_ids = form.provider_user.provider_permissions.pluck(:provider_id)
+      form
+    end
+
+    # TODO: replace with https://github.com/DFE-Digital/apply-for-teacher-training/pull/2553
+    def possible_providers
+      Provider.where(
+        id: current_provider_user.provider_permissions.where(manage_users: true).select(:provider_id),
+      )
+    end
+
+    def persisted?
+      true
+    end
+
+    def save
+      return unless valid?
+
+      not_selected_providers = possible_providers - selected_providers
+
+      selected_providers.each do |provider|
+        ProviderPermissions.find_or_create_by!(provider: provider, provider_user: provider_user)
+      end
+
+      not_selected_providers.each do |provider|
+        permission = ProviderPermissions.find_by(provider: provider, provider_user: provider_user)
+        permission.destroy! if permission
+      end
+
+      true
+    end
+
+    def at_least_one_provider_is_selected
```

Comment:
I don't understand this validation. The user in question may belong to providers A,B,C and the current user (then one with `manage_users`) may belong to B, C, D. The current user will see providers B,C,D in the UI, why shouldn't they be allowed to remove the user in question from both B and C?

Reply:
From our meeting earlier, I think we agreed that this would be handled by an explicit "delete" action, as removing the user from all providers the current user could manage users for would cause them to disappear from the manage-users UI?

Reply:
Ok thanks, makes sense for this iteration.